monai/fl/client/client_algo.py

```diff
@@ -20,60 +21,88 @@ class ClientAlgo(abc.ABC):
     To define a new algo script, subclass this class and implement the
     following abstract methods:
 
-        - #Algo.train()
-        - #Algo.get_weights()
-        - #Algo.evaluate()
+        - self.train()
+        - self.get_weights()
+        - self.evaluate()
 
-    initialize() and finalize() can be optionally be implemented to help with lifecycle management of the object.
+    initialize(), abort(), and finalize() can be optionally be implemented to help with lifecycle management
+    of the class object.
     """
 
-    def initialize(self, extra=None):
-        """call to initialize the ClientAlgo class"""
+    def initialize(self, extra: Optional[dict] = None):
+        """
+        Call to initialize the ClientAlgo class
+
+        Args:
+            extra: optional extra information, e.g. dict of `ExtraItems.CLIENT_NAME` and/or `ExtraItems.APP_ROOT`
+        """
         pass
 
-    def finalize(self, extra=None):
-        """call to finalize the ClientAlgo class"""
+    def finalize(self, extra: Optional[dict] = None):
+        """
+        Call to finalize the ClientAlgo class
+
+        Args:
+            extra: optional extra information
+        """
         pass
 
-    def abort(self, extra=None):
-        """call to abort the ClientAlgo training or evaluation"""
+    def abort(self, extra: Optional[dict] = None):
+        """
+        Call to abort the ClientAlgo training or evaluation
+
+        Args:
+            extra: optional extra information
+        """
+
         pass
 
     @abc.abstractmethod
-    def train(self, data: ExchangeObject, extra=None) -> None:
+    def train(self, data: ExchangeObject, extra: Optional[dict] = None) -> None:
         """
-        objective: train network and produce new network from train data.
-        # Arguments
-        data: ExchangeObject containing current network weights to base training on.
+        Train network and produce new network from train data.
+
+        Args:
+            data: ExchangeObject containing current network weights to base training on.
+            extra: optional extra information
+
+        Returns:
+            None
         """
-        raise NotImplementedError
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
```

Comment:
I wonder whether this error will be ever seen as method is decorated with `@abc.abstractmethod` ?
As we can't instantiate `ClientAlgo`, the only way as far as I can see to raise this error is to call the method directly as static : `ClientAlgo.train(None, None)`...

Reply:
Hi @vfdev-5 ,

That's a good point.
I removed the decorator in the latest commit, it's same as the `Workflow` class now:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/engines/workflow.py#L280
And tested as below:
```py
>>> from monai.fl.client.client_algo import ClientAlgo
>>>
>>>
>>> class Test(ClientAlgo):
...     pass
...
>>> t = Test()
>>> t.train(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/workspace/data/medical/MONAI/monai/fl/client/client_algo.py", line 71, in train
    raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
NotImplementedError: Subclass Test must implement this method.
```

Thanks.

Reply:
I see. So, in this case is there a reason to derive `ClientAlgo` from `abc.ABC` ?

Reply:
You are right, also removed `ABC`.

Thanks.